src/btree/bt_page.c

```diff
@@ -570,7 +572,9 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page, bool check_unstable)
              * Simple keys without compression (not Huffman encoded or prefix compressed), can be
              * directly referenced on the page to avoid repeatedly unpacking their cells.
              */
-            if (!btree->huffman_key && unpack.prefix == 0)
+            if (!btree->huffman_key && unpack.prefix == 0 && unpack.stop_txn == WT_TXN_MAX &&
+              unpack.stop_ts == WT_TS_MAX && txn_global->has_oldest_timestamp &&
+              unpack.start_ts < txn_global->oldest_timestamp)
```

Comment:
Can we ignore items where the `stop_ts < oldest_timestamp`?

Reply:
I prefer not to ignore these when reading, if all entries are obsolete, then we will have an empty page ??
My preference is to ignore obsolete entries as part of reconciliation.

Reply:
It's a performance cost on every search operation. I don't necessarily disagree with you, but it can cost us?

Reply:
@keithbostic Do we need to know ahead of time so we can skip the key cell also (as opposed to just the value)?

Reply:
I've created WT-5471 to track this discussion.

Reply:
We just back up in that case, there are examples in the function:
```
        case WT_CELL_VALUE_OVFL:
            /* Optionally skip unstable values */
            if (check_unstable && __unstable_skip(session, page->dsk, &unpack)) {
                --rip;
                --page->entries;
            }
```